src/benchmarks/gc/src/exec/GCPerfSim/GCPerfSim.cs

```diff
@@ -419,24 +503,43 @@ class SimpleRefPayLoad
     public byte[] payload;
     GCHandle handle;
 
-    // Object header is 2 words.
-    // pointer to 'payload' and handle are both 1 word.
-    // 'payload' itself will have an overhead of 3 words.
-    public static readonly uint Overhead = (2 + 1 + 1 + 3) * Util.POINTER_SIZE;
+    public static readonly uint FieldSize = 2 * Util.POINTER_SIZE;
+    public static readonly uint SimpleRefPayLoadSize = Util.OBJECT_HEADER_SIZE + FieldSize;
+
+    public static uint ArrayOverhead = SimpleRefPayLoadSize + Util.ARRAY_HEADER_SIZE;
+    public static uint SohOverhead = SimpleRefPayLoadSize;
 
     public SimpleRefPayLoad(uint size, bool isPinned, bool isPoh)
     {
-        uint sizePayload = size - Overhead;
+#if STATISTICS
+        Statistics statistics = Statistics.GetStatistics();
+        statistics.sohAllocatedBytes += ReferenceItemWithSize.SohOverhead;
```

Comment:
this one is still not changed?

Reply:
This is intentional.

For each allocation when `allocType=reference`, we create an instance of `ReferenceItemWithSize`, an instance of `SimpleRefPayLoad`, and a byte array. The number `ReferenceItemWithSize.SohOverhead` includes both the size of `ReferenceItemWithSize` and `SimpleRefPayLoad`, so it is the right number of bytes allocated on the SOH for these two objects.

I could have made it such that I account for the size of `ReferenceItemWithSize` in the constructor of `ReferenceItemWithSize` and then account for the size of `SimpleRefPayLoad` in the constructor of `SimpleRefPayLoad`, but then I will need to add one more thread static access and one more addition in the constructor `ReferenceItemWithSize`. This can be seen as an optimization so that I saved those operations.